x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -115,12 +118,6 @@ export class TaskRunnerFactory {
         const request = getFakeRequest(apiKey);
         basePathService.set(request, path);
 
-        // TM will treat a task as a failure if `attempts >= maxAttempts`
```

Comment:
Task runner no longer cares if the task is retried or not. It will throw the appropriate error, log errors messages and leave it up to Task Manager to determine if it's done retrying or not.